packages/@expo/metro-config/src/serializer/withExpoSerializers.ts

```diff
@@ -62,17 +64,113 @@ export function withSerializerPlugins(
   };
 }
 
+export function createDefaultExportCustomSerializer(config: Partial<MetroConfig>): Serializer {
+  return async (
+    entryPoint: string,
+    preModules: readonly Module<MixedOutput>[],
+    graph: ReadOnlyGraph<MixedOutput>,
+    options: SerializerOptions<MixedOutput>
+  ): Promise<string | { code: string; map: string }> => {
+    const isPossiblyDev = graph.transformOptions.hot;
+    // TODO: This is a temporary solution until we've converged on using the new serializer everywhere.
+    const enableDebugId = options.inlineSourceMap !== true && !isPossiblyDev;
+
+    let debugId: string | undefined;
+    const loadDebugId = () => {
+      if (!enableDebugId || debugId) {
+        return debugId;
+      }
+
+      // TODO: Perform this cheaper.
+      const bundle = baseJSBundle(entryPoint, preModules, graph, {
+        ...options,
+        debugId: undefined,
+      });
+      const outputCode = bundleToString(bundle).code;
+      debugId = stringToUUID(outputCode);
+      return debugId;
+    };
+
+    let bundleCode: string | null = null;
+    let bundleMap: string | null = null;
+
+    if (config.serializer?.customSerializer) {
+      const bundle = await config.serializer?.customSerializer(
+        entryPoint,
+        preModules,
+        graph,
+        options
+      );
+      if (typeof bundle === 'string') {
+        bundleCode = bundle;
+      } else {
+        bundleCode = bundle.code;
+        bundleMap = bundle.map;
+      }
+    } else {
```

Comment:
The default serializer is not created if there is an existing custom serializer. -> https://github.com/expo/expo/pull/25935/files#diff-a07d4017951693f251b9c918dec8b9f2a2d2896f77d212181e6d98444050eeceR172

The custom serializer might not generate a debug ID we should not try to fill in the map and generate the ID for it. As it might be loose not matching any bundle.

I'm proposing this: https://github.com/expo/expo/compare/%40evanbacon/metro-config/debug-id-support...krystofwoldrich:expo:kw-add-debug-id-to-default-serializer

Reply:
Seems fine, but the sentry plugin would possibly not use this as it would only apply to one possible export path, `export:embed`. If we expose a more limited function like `modifyDeps(preModules, { debugId })` then it will work in both code paths.